cmd/agent/api/server.go

```diff
@@ -44,14 +35,22 @@ import (
 	"github.com/DataDog/datadog-agent/pkg/api/util"
 	"github.com/DataDog/datadog-agent/pkg/config"
 	remoteconfig "github.com/DataDog/datadog-agent/pkg/config/remote/service"
-	pb "github.com/DataDog/datadog-agent/pkg/proto/pbgo/core"
-	"github.com/DataDog/datadog-agent/pkg/tagger"
-	taggerserver "github.com/DataDog/datadog-agent/pkg/tagger/server"
-	grpcutil "github.com/DataDog/datadog-agent/pkg/util/grpc"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
 	"github.com/DataDog/datadog-agent/pkg/util/optional"
 )
 
-var listener net.Listener
+func startServer(listener net.Listener, srv *http.Server, name string) {
+	// Use a stack depth of 4 on top of the default one to get a relevant filename in the stdlib
+	logWriter, _ := config.NewLogWriter(5, seelog.ErrorLvl)
+
+	srv.ErrorLog = stdLog.New(logWriter, fmt.Sprintf("Error from the agent http %s: ", name), 0) // log errors to seelog
+
+	tlsListener := tls.NewListener(listener, srv.TLSConfig)
+
+	go srv.Serve(tlsListener) //nolint:errcheck
+
+	log.Infof("%s started on %s", name, listener.Addr().String())
+}
 
 // StartServer creates the router and starts the HTTP server
 func StartServer(
```

Comment:
maybe a nit
```suggestion
func StartAPIServers(
```

Reply:
Considering it's in a package named `api`, I just renamed it `StartServers`